src/adv.py

```diff
@@ -32,20 +40,63 @@
 room['narrow'].w_to = room['foyer']
 room['narrow'].n_to = room['treasure']
 room['treasure'].s_to = room['narrow']
+room['treasure'].d_to = room['stairway']
+room['stairway'].u_to = room['treasure']
+room['stairway'].n_to = room['secretroom']
+room['secretroom'].s_to = room['stairway']
+room['secretroom'].k_to = room['realtreasureroom']
+
+player = Player(room['outside'])
+
+def direction_successful(direction, current_room):
+    attribute = direction + "_to"
+
+    if hasattr(current_room, attribute):
+        return getattr(current_room, attribute)
+    elif(attribute == "q_to"):
+        return
+    else:
+        print("You can't go that way")
+        return current_room
+
+while True:
+    print(player.current_room.name)
+    print(player.current_room.description)
+    print(player.current_room.equipment)
+
+    player_input = input("\n>").lower().split()
+    
+    if len(player_input) == 1:
+        player_input = player_input[0][0]    
+        player.current_room = direction_successful(player_input, player.current_room)
```

Comment:
instead of first trying to move, and then having the "q_to" clause in your move function,  you could check player_input[0] for the command and decide what to do from there.
like:
  if player_info[0] == 'q'  -> do the quit stuff
else if player_info[0] in ['n','s','e','w'] -> do the move stuff
else if player_info[0] == 'take' -> do the grab item stuff